### PR TITLE
[SDESK-4790] - Shortcut buttons checking user privileges

### PR DESCRIPTION
**Pull request by akintolga**
- Also addressing an issue in `To Desk` shortcut that moves the story to the next stage

### CODE REVIEW

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.ts [hunk 1]
```diff
@@ -171,7 +171,8 @@ export function AuthoringDirective(
              * @returns {Boolean}
              */
             $scope.canPublishOnDesk = function() {
-                return !($scope.deskType === 'authoring' && config.features.noPublishOnAuthoringDesk);
+                return !($scope.deskType === 'authoring' && config.features.noPublishOnAuthoringDesk) &&
+                    !!$scope.privileges.publish;
```
Comment (tomaskikutis) [inline scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: Is `!!$scope.privileges.publish` meant to mean `$scope.privileges.publish != null`? Use the explicit version.

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.ts [hunk 2]
```diff
@@ -862,8 +863,9 @@ export function AuthoringDirective(
             };
 
             $scope.sendToNextStage = function() {
-                var stageIndex, stageList = desks.deskStages[desks.activeDeskId];
-                var selectedStage, selectedDesk = desks.deskLookup[desks.activeDeskId];
+                var currentDeskId = desks.getCurrentDeskId() || (userDesks && userDesks[0]);
```
Comment (tomaskikutis) [inline scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: What's the change here? Does something not work with using `activeDeskId`? Why is fallback to `userDesks[0]` used?

File: scripts/apps/authoring/views/authoring-topbar.html [hunk 3]
```diff
@@ -58,28 +58,25 @@
...
                 type="submit"
-                ng-if="canPublishOnDesk()"
                 ng-click="saveAndContinue(publish, true)"
                 title="{{ :: 'Publish' | translate }}">
                 <span class="btn__text" translate>Publish</span>
                 <span class="btn__text--short" translate>P</span>
         </button>
         <button
-                ng-if="$root.config.features.customAuthoringTopbar.publishAndContinue === true"
+                ng-if="$root.config.features.customAuthoringTopbar.publishAndContinue === true && canPublishOnDesk() && canRewriteArticle()"
```
Comment (tomaskikutis) [inline scripts/apps/authoring/views/authoring-topbar.html]: Why is `canRewriteArticle` checked for `Publish and continue` action, but not for `Publish`?

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.ts (hunk 2 above)
Comment (akintolga) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: I think `getCurrentDeskId` uses `activeDeskId` so it is better to use the `getCurrentDeskId` here. It also considers users desks as well, so I removed some of the redundant bits there. Also added an error message where there is no current desk id (esp if the user is working in his personal space).

File: scripts/apps/authoring/views/authoring-topbar.html (hunk 3 above)
Comment (akintolga) [reply scripts/apps/authoring/views/authoring-topbar.html]: `Publish and continue` publishes a story and creates an update right after, That's why it requires `canRewriteArticle` permission.

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.ts [hunk 4]
```diff
@@ -862,8 +863,15 @@ export function AuthoringDirective(
             };
 
             $scope.sendToNextStage = function() {
-                var stageIndex, stageList = desks.deskStages[desks.activeDeskId];
-                var selectedStage, selectedDesk = desks.deskLookup[desks.activeDeskId];
+                var currentDeskId = desks.getCurrentDeskId();
+
+                if (_.isNil(currentDeskId)) {
+                    notify.error(gettext('Failed to send to next stage.'));
```
Comment (tomaskikutis) [inline scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: When is `currentDeskId` null/undefined? Is it a valid use case or an error? If it's an error, add a `logger.warn` call so the error gets sent to sentry and we can find and fix it.

Comment (akintolga) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: currentDeskId could be null if the user is working on his personal space so I don't think it is an error

Comment (tomaskikutis) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: Can you add a reason to the error message so it's clear to the user why the action doesn't work?

**Pull request merged**